### ISSUE TITLE
bug:  Cursor position outside buffer

### ISSUE THREAD

**Issue by 0x67cq** [folke/trouble.nvim#329]
### Did you check docs and existing issues?

- [X] I have read all the trouble.nvim docs
- [X] I have searched the existing issues of trouble.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.9.0

### Operating system/version

MacOS 13.4.1

### Describe the bug

function() require("trouble").open("lsp_references"), then press  <c-v>    print below erro stack

E5108: Error executing lua ....local/share/nvim/lazy/trouble.nvim/lua/trouble/util.lua:28: Cursor position outside buffer
stack traceback:
        [C]: in function 'nvim_win_set_cursor'
        ....local/share/nvim/lazy/trouble.nvim/lua/trouble/util.lua:28: in function 'jump_to_item'
        ....local/share/nvim/lazy/trouble.nvim/lua/trouble/view.lua:477: in function 'jump'
        ....local/share/nvim/lazy/trouble.nvim/lua/trouble/init.lua:197: in function 'action'
        [string ":lua"]:1: in main chunk

this is error stack. and  I try to print some log. look like this:
```
    print("bufnr", item.bufnr, "win", win, "item.start.line", item.start.line, "character", item.start.character)
    vim.api.nvim_set_current_buf(item.bufnr)
    vim.api.nvim_win_set_cursor(win or 0, { item.start.line + 1, item.start.character })

    bufnr 4 win 1000 item.start.line 98 character 11

    print(opts.win, opts.item, opts.precmd)
    print(self.parent)
    util.jump_to_item(opts.win or self.parent, opts.precmd, item)

    nil nil vsplit
    1000
```
then I add one line such as:
```
    print("bufnr", item.bufnr, "win", win, "item.start.line", item.start.line, "character", item.start.character)
    local win = vim.api.nvim_get_current_win()
    print("bufnr", item.bufnr, "win", win, "item.start.line", item.start.line, "character", item.start.character)
    vim.api.nvim_set_current_buf(item.bufnr)
    
    nil nil vsplit
    1000
    bufnr 4 win 1000 item.start.line 98 character 11
    bufnr 4 win 1003 item.start.line 98 character 11
```
and it work. I'm a beginner in nvim plugin. I can't descript it. but I think it's because self.parent windows ID not equal current_win and can't usr vim.api.nvim_win_set_cursor to jump. like view.lua's View object's parent in func jump hasn't assignment.

### Steps To Reproduce

1. vim.keymap.set("n", "<leader>gr", function() require("trouble").open("lsp_references") end)
2. golang project and call this func
3. print error stack

### Expected Behavior

when I use trouble open lsp_references, and press c-v, it will open a vsp window and show references 

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/trouble.nvim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here
```

**Comment by folke** [comment]
Development on the main branch is EOL.

Trouble has been rewritten and will be merged in main soon.

This issue/feature either no longer exists or has been implemented on dev.

For more info, see https://github.com/folke/trouble.nvim/tree/dev

**Comment by pwang2** [comment]
I can verify this still an issue for typescript source map scenario.   I am muting it in Noice as a workaround. 

Quite a few language servers implementation can be quite quick buggy and inconsistent.  I think we can tolerate when lsp jump not working as expected.

**Comment by folke** [comment]
@pwang2 first of all, make sure you update trouble, since I've been making changes today.

Op's issue can no longer exist since Trouble has been fully rewritten.

If you have a similar issue, please open a new issue with the steps to reproduce and the exact error message you get.

**Comment by pwang2** [comment]
thanks for your great work and quick response.  I have updated to commit 46cf952 already and it is still the case. 
![image](https://github.com/user-attachments/assets/07f5ec68-1b2d-4c71-bcb7-7dc0cea8a575)

**Comment by folke** [comment]
Showing me the exact error would help

**Comment by pwang2** [comment]
The error here is the lsp jump target the file here `ui-base-components/src/components/DashboardCard.vue` but none of my file has more than 269 lines.    

File jump is okay but the line position is not correct. I guess it can be from the transpiling from vue but I failed to find a file more than 269 line in tranpilated files. 

```lua
{
  basename = "DashboardCard.vue",
  buf = 3,
 
  dirname = "/home/pwang2/ui-core/packages/ui-base-components/src/components",
  end_pos = { 269, 0 },
  filename = "/home/pwang2/ui-core/packages/ui-base-components/src/components/DashboardCard.vue",
  item = {
    client = "ts_ls",
    client_id = 2,
    location = {
      originSelectionRange = {
        ["end"] = {
          character = 18,
          line = 2
        },
        start = {
          character = 5,
          line = 2
        }
      },
      targetRange = {
        ["end"] = {
          character = 0,
          line = 268
        },
        start = {
          character = 0,
          line = 268
        }
      },
      targetSelectionRange = {
        ["end"] = {
          character = 0,
          line = 268
        },
        start = {
          character = 0,
          line = 268
        }
      },
      targetUri = "file:///home/pwang2/ui-core/packages/ui-base-components/src/components/DashboardCard.vue"
    },
    text = ""
  },
  pos = { 269, 0 },
  source = "lsp"
}
```

**Comment by pwang2** [comment]
I can confirm it is related to the vue-language-server(volar)'s hybridMode.   
https://gist.github.com/johnsoncodehk/62580d04cb86e576e0e8d6bf1cb44e73

Actually, the behavior of Trouble lsp_definitions are quite differently.  
hybridMode = false:  trouble shows as:
![image](https://github.com/user-attachments/assets/5b4e2b1c-1b43-4057-be5a-96236dfd0964)

hybridMode enabled.  The actually file jump proceed.   but the jump positions are randomly incorrect, based on the language server attach order... 

TLDR
- the issue I saw here is not `trouble.nvim` related.     
- as a workaround, we can either mute the error prompt in noice.nvim if we use.  or, setting hybridMode to false and result to a selection jump instead.
